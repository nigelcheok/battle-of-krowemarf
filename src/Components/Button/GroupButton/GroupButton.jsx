import React from 'react';

export function GroupButton(props) {
  return (
    <div className="btn-group" role="group" aria-label="" style={ GroupButtonStyle() }>
      <button type="button" className="btn btn-outline-secondary font-weight-bold" disabled>
        <div className={`fa ${props.icon} fa-1x`} style={GroupButtonIconStyle()}/>
        { props.label }
      </button>
      <button type="button" className="btn btn-outline-secondary font-weight-bold" disabled>{ props.value }</button>
    </div>
  );
}

function GroupButtonStyle() {
  return {
    marginRight: '12px',
  }
}

function GroupButtonIconStyle() {
  return {
    display: 'inline-block',
    marginRight: '6px',
  }
}
