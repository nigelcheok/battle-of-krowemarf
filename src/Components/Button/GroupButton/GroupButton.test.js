import React from 'react';
import { shallow } from 'enzyme';
import { GroupButton } from './GroupButton';

describe('GroupButton', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<GroupButton/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with label and value', () => {
    const component = shallow(<GroupButton label="test" value="value description"/>);
    expect(component).toMatchSnapshot();
  });
});