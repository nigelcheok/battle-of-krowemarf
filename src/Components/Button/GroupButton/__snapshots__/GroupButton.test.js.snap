// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupButton should render correctly with label and value 1`] = `
<div
  className="display-inline-block mt-2 mt-lg-0"
>
  <div
    aria-label=""
    className="btn-group"
    role="group"
    style={
      Object {
        "marginRight": "12px",
      }
    }
  >
    <button
      className="btn btn-outline-transparent font-weight-bold"
      disabled={true}
      style={
        Object {
          "pointerEvents": "none",
        }
      }
      type="button"
    >
      <div
        className="fa undefined fa-1x"
        style={
          Object {
            "display": "inline-block",
            "marginRight": "6px",
          }
        }
      />
      test
    </button>
    <button
      className="btn btn-outline-transparent"
      disabled={true}
      style={
        Object {
          "pointerEvents": "none",
        }
      }
      type="button"
    >
      value description
    </button>
  </div>
</div>
`;

exports[`GroupButton should render correctly with no props 1`] = `
<div
  className="display-inline-block mt-2 mt-lg-0"
>
  <div
    aria-label=""
    className="btn-group"
    role="group"
    style={
      Object {
        "marginRight": "12px",
      }
    }
  >
    <button
      className="btn btn-outline-transparent font-weight-bold"
      disabled={true}
      style={
        Object {
          "pointerEvents": "none",
        }
      }
      type="button"
    >
      <div
        className="fa undefined fa-1x"
        style={
          Object {
            "display": "inline-block",
            "marginRight": "6px",
          }
        }
      />
    </button>
    <button
      className="btn btn-outline-transparent"
      disabled={true}
      style={
        Object {
          "pointerEvents": "none",
        }
      }
      type="button"
    />
  </div>
</div>
`;
